makefiles/tools/stm32programmer.inc.mk

```diff
@@ -0,0 +1,27 @@
+FLASHFILE ?= $(ELFFILE)
+
+STM32_PROGRAMMER_CLI_PATH ?=
+ifeq (,$(STM32_PROGRAMMER_CLI_PATH))
+  $(error "STM32_PROGRAMMER_CLI_PATH is unset")
+endif
+
+STM32_PROGRAMMER_CLI_MODE ?= STM32_UART_BOOLOADER
+ifneq (,$(filter STM32_UART_BOOLOADER,$(STM32_PROGRAMMER_CLI_MODE)))
+  PROG_BAUD ?= 115200
+  STM32_PARTITION_ID ?= 0x00
+  STM32_PROGRAMMER_CLI_FLAGS += --connect port=$(PROG_DEV) br=$(PROG_BAUD)
+  STM32_PROGRAMMER_CLI_FLAGS += --write $(FLASHFILE) $(STM32_PARTITION_ID)
+else
+  ifneq (,$(filter STM32_STLINK,$(STM32_PROGRAMMER_CLI_MODE)))
```

Comment:
by default, that won't work with boards having an st-link programmer, which the majority. Would it make sense to set `STM32_STLINK` as the default mode, and set `STM32_UART_BOOTLOADER` in `lora-e5-dev`'s `Makefile.include` ? Maybe other boards could be adapted as well ?